arrays/arrays.c

```diff
@@ -34,12 +35,19 @@ Array *create_array (int capacity) {
 /*****
  * Free memory for an array and all of its stored elements
  *****/
-void destroy_array(Array *arr) {
+void destroy_array(Array *arr) 
+{
```

Comment:
As it is now, you're freeing the arr->elements array which contain pointers pointing to pointers pointing to the strings you want stored and the arr struct, but you're not freeing the pointers pointing to the strings. Without that additional freeing step, the string pointers will be left taking up space in your memory (i.e. a memory leak). These string pointers are created when you copy the input string in the append and insert functions (string should be copied via strdup() in these functions).